src/tools/server-autosave.js

```diff
@@ -0,0 +1,18 @@
+import {
+  setPreviouslySavedJsmdToCurrentJsmd,
+  saveNotebookToServer
+} from "../actions/actions";
+
+export function changeMadeToJsmd(state) {
```

Comment:
could be `notebookChangedSinceSave` or `notebookChangedSinceUpload` -- more generic to allow comparison of other notebook stuff